src/meta-srv/src/procedure/region_failover/deactivate_region.rs

```diff
@@ -42,6 +43,35 @@ impl DeactivateRegion {
         Self { candidate }
     }
 
+    async fn mark_leader_downgraded(
+        &self,
+        ctx: &RegionFailoverContext,
+        failed_region: &RegionIdent,
+    ) -> Result<()> {
+        let table_id = failed_region.table_id;
+
+        let table_route_value = ctx
+            .table_metadata_manager
+            .table_route_manager()
+            .get(table_id)
+            .await
+            .context(error::TableMetadataManagerSnafu)?
+            .context(error::TableRouteNotFoundSnafu { table_id })?;
+
```

Comment:
It's inevitable we still need one additional get request. In the future, the migration procedure will store these data in a volatile context to avoid this issue and fetch from the remote only if the volatile context is empty(e.g., after recovering).